openedx/core/djangoapps/api_admin/management/commands/create_api_access_request.py

```diff
@@ -0,0 +1,130 @@
+""" Management command to create an ApiAccessRequest for given users """
+import logging
+from django.core.management.base import BaseCommand, CommandError
+from django.contrib.auth.models import User
+from django.contrib.sites.models import Site
+
+from openedx.core.djangoapps.api_admin.models import (
+    ApiAccessConfig,
+    ApiAccessRequest,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Create an ApiAccessRequest for the given user
+
+    Example usage:
+        $ ./manage.py lms create_api_request <username> --create-config
+    """
+
+    help = 'Create an ApiAccessRequest for the given user'
+    DEFAULT_STATUS = ApiAccessRequest.APPROVED
+    DEFAULT_WEBSITE = 'www.test-edx-example-website.edu'
+    DEFAULT_REASON = 'Generated by management job create_api_request'
+
+    def add_arguments(self, parser):
+        parser.add_argument('username', type=str)
+        parser.add_argument(
+            '--create-config',
+            action='store_true',
+            help='Create ApiAccessConfig if it does not exist'
+        )
+        parser.add_argument(
+            '--status',
+            type=str,
+            choices=[choice[0] for choice in ApiAccessRequest.STATUS_CHOICES],
+            help='Status of the created ApiAccessRequest'
+        )
+        parser.add_argument(
+            '--reason',
+            type=str,
+            help='Reason of the created ApiAccessRequest'
+        )
+        parser.add_argument(
+            '--website',
+            type=str,
+            help='Website of the created ApiAccessRequest'
+        )
+        parser.add_argument(
+            '--site-name',
+            type=str,
+            help='Name of an existing Site that should be associated with this ApiAccessRequest'
+        )
+
+    def handle(self, *args, **options):
+        user = self.get_user(options.get('username'))
+        site = self.get_site(options.get('site_name'))
+        self.create_api_access_request(
+            user,
+            options.get('status'),
+            options.get('reason'),
+            options.get('website'),
+            site,
+        )
+        if options.get('create_config'):
```

Comment:
We can create the api_access_request before we create the config?
Logically, it makes more sense if we create_if_necessary on the config first. So my suggestion is to move these two lines above line 60